### DISCUSSION TITLE
Does broadcasting to the world the platforms being used compromise security?

### DISCUSSION THREAD

**Discussion by MPatriot** [oqtane/oqtane.framework#3237] [General] [answered]
Most that look at their logs will notice an occasional flurry of phishing type requests made on your websites.

Most of these requests are apparent guesses at typical Word Press admin paths and other popular hosting environments.

Anyone using Oqtane is required to post the MIT License on their site and I assume show a link so it can be found...

This of course is only right.  Most of us that use Oqtane for free with no contributions, and actually use more of the contributors time by asking sometimes obvious questions or questions that arise by reasons not originating with Oqtane or the users lack of knowledge or understanding(often the case with me.)

It is actually biblical and or just moral that a workman is worthy of his wages.  I myself have found Oqtane to not only help making multi-user environments more RAD but very educational in how this whole Blazor thing works and I thank the contributors. It is the least of things I can do to help promote the effort.

The question...finally, is this:

Does blatantly telling the world the exact platforms sites use make it more susceptible to hacking? Though I'm pretty sure anyone competent could figure it out anyway.

**Comment by sbwalker** [comment]
@MPatriot where did you get the information that the MIT license must be published on an Oqtane site? This is absolutely false. The framework has no requirement that a user needs to publish the copyright notice or license.  I have actually never seen an Oqtane site publish this information - except for https://www.oqtane.org (which is for informational purposes so that potential users know which open source license it is available under). I am very confused by this post and would appreciate if you could let me know where this false information originated.

**Reply by MPatriot** [reply]
@sbwalker Really?  I don't recall where I got that idea.  Perhaps watching you on one of your You Tube videos?
I thought I heard you say somewhere that if someone is using Oqtane, even the smallest amount that publishing the license agreement on the site is a requirement.
If, where or when I got that idea in my mind I don't know.  Perhaps I misheard, misread something somewhere.  I certainly didn't get the idea recently but a long time ago.  
Thanks for the clarification and all of your hard work with others!

**Reply by MPatriot** [reply]
I would still be interested in your opinion if publishing information on a sites platform ASP.Net and / or Oqtane exposes or makes it more obvious of possible security holes?

-Mark

**Reply by sbwalker** [reply, marked as answer]
@MPatriot I have definitely never said that attribution is required for Oqtane. The framework does not inject an identifier,  copyright notice, or license into any of its run-time output. And users do not need to disclose the fact they are using Oqtane on their websites or web applications. So wherever this idea originated, it is 100% false.

In regards to security... it is generally a good idea to try and obfuscate the technical details of your infrastructure. However this is not always a simple task as most infrastructure has unique fingerprints which can be used to identify their presence. If you are familiar with [BuiltWith.Com](https://builtwith.com/), this is exactly how it is able to determine the infrastructure for any public website. And in theory if a malicious was able to determine that you were using a specific component which contained a vulnerability, then obviously they could try to exploit it.  However, I don't think this question is specific to Oqtane...

**Reply by MPatriot** [reply]
@sbwalker  Thanks for the clarifications and BTW, thank you for modifying the registration regarding punctuation.

**Reply by leigh-pointer** [reply]
@MPatriot the first version was a headache! Then yours truly mr @sbwalker made it more slik!

**Reply by MPatriot** [reply]
@leigh-pointer haha! You guys are great!  
You once asked me why I was using debug mode...its was kinda simple then, though I don't think I answered:

I needed stopping points to trace through the maze of it all and get a basic understanding of what the heck is going on!

Thanks:)

**Reply by MPatriot** [reply]
Ohh, I found it.  Starring right at me everytime I do a default setup:

MIT License
Copyright (c) 2018-2023 .NET Foundation

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

**The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.**

I may be misreading that but I certainly wasn't just making stuff up either:)

I suppose that actually refers to the distribution of the code and not the use or implementation of it?

**Reply by sbwalker** [reply]
@MPatriot the MIT license is the most commonly used open source license worldwide. The attribution clause you highlighted basically says that you cannot remove the copyright notice from the source code to hide its original or claim that you wrote the application yourself. It does not say that you need to prominently display the copyright notice in public facing implementations.

**Reply by vnetonline** [reply]
Plus no where does it mention Oqtane in the above license

**Reply by MPatriot** [reply]
Guys, I get it!
I would not and was not complaining if it was a requirement,  
It was asked where I could possibly have come up with such an idea and I saw something right in front of my face that I have seen endless amount of times but apparently didn't read all that closely.

I have been wrong about most things or assumptions I have made here, especially the email thing.  I turns out Shaun was right about that all along also...Google turns off "Less Secure Applications" permissions when they haven't been used in a while.

Nothing in this post was meant to complain about anything and I am sorry if it came across that way.  Just a follow up on how I could have come up with such an idea in the first place.  I was in fact laughing at myself and probably should have kept it to myself.

I am in fact very thankful and grateful to have such a well rounded example and template for what can be developed into a full blown scalable application within what I consider the environment of the future today.

Again, thanks for all of your hard work.

Mark

**Reply by sbwalker** [reply]
@MPatriot no worries... I am actually glad you brought this up, as its much better to provide clarity than to have people make any false assumptions.
